packages/react-instantsearch/src/core/createInstantSearchServer.js

```diff
@@ -0,0 +1,136 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { version } from '../../package.json';
+import algoliasearchHelper, {
+  SearchResults,
+  SearchParameters,
+} from 'algoliasearch-helper';
+import ReactDom from 'react-dom/server';
+import { getIndex, hasMultipleIndex } from './indexUtils';
+import { isEmpty } from 'lodash';
+import cis from './createInstantSearch';
+
+const searchParameters = [];
+let client;
+let indexName = '';
+
+const onSearchParameters = function(
+  getSearchParameters,
+  context,
+  props,
+  searchState
+) {
+  const index = getIndex(context);
+  searchParameters.push({
+    getSearchParameters,
+    context,
+    props,
+    searchState,
+    index,
+  });
+};
+
+const findResults = function(App, params) {
+  ReactDom.renderToString(<App {...params} />);
```

Comment:
Let's try to access client and indexName here:

const element = <App {...params} />;
console.log(element.client/indexName)